pyserini/dsearch/_dsearcher.py

```diff
@@ -366,6 +369,28 @@ def load_index(self, index_dir: str):
         docids = self.load_docids(docid_path)
         return index, docids
 
+    def doc(self, docid: Union[str, int]) -> Optional[Document]:
+        """Return the :class:`Document` corresponding to ``docid``. Since dense indexes don't store documents
+        but sparse indexes do, route over to corresponding sparse index (according to prebuilt_index_info.py)
+        and use its doc API 
+
+        Parameters
+        ----------
+        docid : Union[str, int]
+            Overloaded ``docid``: either an external collection ``docid`` (``str``) or an internal Lucene ``docid``
+            (``int``).
+
+        Returns
+        -------
+        Document
+            :class:`Document` corresponding to the ``docid``.
+        """
+        if self.prebuilt_index_name is None:
+            return None
+        sparse_index = get_sparse_index(self.prebuilt_index_name)
+        ssearcher = SimpleSearcher.from_prebuilt_index(sparse_index)
```

Comment:
this ssearch initialization should be put into constructor?

Reply:
I wasn't sure if this would become an unnecessary overhead in the constructor, but i'll put it in